backend/BjjAppApi/Program.cs

```diff
@@ -12,9 +12,9 @@
 builder.Services.AddSingleton(mongoDatabase.GetCollection<Move>(bjjAppDatabaseSettings.MovesCollectionName));
 builder.Services.AddSingleton(mongoDatabase.GetCollection<Demo>(bjjAppDatabaseSettings.DemosCollectionName));
 builder.Services.AddSingleton(mongoDatabase.GetCollection<Name>(bjjAppDatabaseSettings.NamesCollectionName));
-builder.Services.AddSingleton<MovesService>();
-builder.Services.AddSingleton<DemosService>();
-builder.Services.AddSingleton<NamesService>();
+builder.Services.AddSingleton<IMovesRepository, MovesMongoRepository>();
+builder.Services.AddSingleton<IDemosRepository, DemosMongoRepository>();
+builder.Services.AddSingleton<INamesRepository, NamesMongoRepository>();
```

Comment:
Why do we need to specify an abstract and concrete type here? i.e. `<IMovesRepo, MovesMongoRepository>` shouldn't it just be good enough to do `<IMovesRepo>`? Probably a C# thing but would like to know why.

Reply:
That's what I thought as well but it wasn't good enough to do just `<IMovesRepo>`. I got an exception saying that the service of that type could not be found. So I followed what the [docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-6.0#overview-of-dependency-injection) had.

Reply:
Ahh gotcha, i find no issue with this then.